starcluster/node.py

```diff
@@ -266,12 +256,22 @@ def tags(self):
         return self.instance.tags
 
     def add_tag(self, key, value=None):
-        try:
-            return self.instance.add_tag(key, value)
-        except EC2ResponseError as e:
-            if e.error_code == "InvalidInstanceID.NotFound":
-                raise exception.InstanceDoesNotExist(self.id)
-            raise e
+        count = 0
+        while True:
+            try:
+                try:
+                    return self.instance.add_tag(key, value)
+                except EC2ResponseError as e:
+                    if e.error_code == "InvalidInstanceID.NotFound":
+                        raise exception.InstanceDoesNotExist(self.id)
+                    raise e
+            except exception.InstanceDoesNotExist:
+                count += 1
```

Comment:
Can't we move the block starting at 269 into the first except handling code if the if statement (line 266)?
That way we can use a single try catch instead of two.